docs/application/native/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1134 @@
+  {
+      int ret;
+      ret = vc_mgr_unset_state_changed_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the service state changed callback to get invoked when the Voice control service state changes:
+
+  ```c
+  /* Callback */
+  void
+  __service_state_changed_cb(vc_service_state_e previous, vc_service_state_e current, void* user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_service_state_changed_cb()
+  {
+      int ret;
+      ret = vc_mgr_set_service_state_changed_cb(__service_state_changed_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_service_state_changed_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_service_state_changed_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the current language changed callback to get invoked when the voice control setting language changes:
+
+  ```c
+  /* Callback */
+  void
+  __current_language_changed_cb(const char* previous, const char* current, void* user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_current_language_changed_cb()
+  {
+      int ret;
+      ret = vc_mgr_set_current_language_changed_cb(__current_language_changed_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_current_language_changed_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_current_language_changed_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the error callback to get invoked when an error occurs in the voice control manager process:
+
+  ```c
+  /* Callback */
+  void
+  __error_cb(vc_error_e reason, void* user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_error_cb()
+  {
+      int ret;
+      ret = vc_mgr_set_error_cb(__error_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_error_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_error_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the speech detected callback to get invoked when the beginning of speech or the end of speech is detected:
+
+  ```c
+  /* Callback */
+  void
+  __speech_detected_cb(void* user_data)
+  {
+      /* Your code */
+  }
+
+  /* Set */
+  void
+  set_speech_detected_cb()
+  {
+      int ret;
+      ret = vc_mgr_set_speech_detected_cb(__speech_detected_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Unset */
+  void
+  unset_speech_detected_cb()
+  {
+      int ret;
+      ret = vc_mgr_unset_speech_detected_cb();
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Set the dialog requested callback to get invoked when the voice control manager or engine requests conversation for additional information about the current utterance:
```

Comment:
- Set the dialog requested callback to get invoked when the voice control manager or engine requests conversation for additional information about the current utterance:

What is meant by conversation here? Is it correct in this context? Please verify.

Reply:
For example, if the user says 'play music', the voice control manager app or the voice control service engine may need more information like which app launches to play music. In this case, the manager app or the service engine can request to play and display some text using this callback. That is, the manager app or the service engine can request additional information using Q&A using this callback function.
So, I use coversation because this kind of process(1. user says something 2. manager app asks question for more information 3. user says the answewr) is similar to human coversation.